rskj-core/src/main/java/org/ethereum/vm/program/Program.java

```diff
@@ -1410,6 +1414,14 @@ public void callToPrecompiledAddress(MessageCall msg, PrecompiledContract contra
             contract.init(internalTx, executionBlock, track, this.invoke.getBlockStore(), null, result.getLogInfoList());
         }
 
+        if (contract instanceof PrecompiledContracts.Environment) {
+            PrecompiledContractArgs args = PrecompiledContractArgsBuilder.builder()
+                    .programInvoke(this.invoke)
+                    .build();
+
+            contract.init(args);
```

Comment:
could you please explain why we need this second `init()` call? Can't we just provide the `invoke` instance in the [one](https://github.com/rsksmart/rskj/pull/2173/files#diff-c0381bea471a7d1522a7283019e890ec5e41f74e8eae4ee07eac99d432690d93R1414) above?